### ISSUE TITLE
MQTT related errors in the log

### ISSUE THREAD

**Issue by Fannangir** [OldSmurf56/xpeng#4]
Hi, got the following MQTT errors:

Logger: homeassistant.components.mqtt.entity
Source: components/mqtt/entity.py:155
integration: MQTT (documentation, issues)
First occurred: 9:13:59 AM (242 occurrences)
Last logged: 7:14:00 PM

Error 'expected SensorDeviceClass or one of 'date', 'enum', 'timestamp', 'apparent_power', 'aqi', 'area', 'atmospheric_pressure', 'battery', 'blood_glucose_concentration', 'carbon_monoxide', 'carbon_dioxide', 'conductivity', 'current', 'data_rate', 'data_size', 'distance', 'duration', 'energy', 'energy_storage', 'frequency', 'gas', 'humidity', 'illuminance', 'irradiance', 'moisture', 'monetary', 'nitrogen_dioxide', 'nitrogen_monoxide', 'nitrous_oxide', 'ozone', 'ph', 'pm1', 'pm10', 'pm25', 'power_factor', 'power', 'precipitation', 'precipitation_intensity', 'pressure', 'reactive_power', 'signal_strength', 'sound_pressure', 'speed', 'sulphur_dioxide', 'temperature', 'volatile_organic_compounds', 'volatile_organic_compounds_parts', 'voltage', 'volume', 'volume_storage', 'volume_flow_rate', 'water', 'weight', 'wind_speed' for dictionary value @ data['device_class']' when processing MQTT discovery message topic: 'homeassistant/sensor/xpeng_is_reachable/config', message: '{'state_topic': 'xpeng/is_reachable', 'device_class': 'connectivity', 'unique_id': 'xpeng_is_reachable', 'device': {'identifiers': ['xpeng'], 'model': 'Xpeng Model', 'manufacturer': 'Xpeng', 'name': 'Xpeng'}, 'name': 'Xpeng Is Reachable'}'
Error 'expected SensorDeviceClass or one of 'date', 'enum', 'timestamp', 'apparent_power', 'aqi', 'area', 'atmospheric_pressure', 'battery', 'blood_glucose_concentration', 'carbon_monoxide', 'carbon_dioxide', 'conductivity', 'current', 'data_rate', 'data_size', 'distance', 'duration', 'energy', 'energy_storage', 'frequency', 'gas', 'humidity', 'illuminance', 'irradiance', 'moisture', 'monetary', 'nitrogen_dioxide', 'nitrogen_monoxide', 'nitrous_oxide', 'ozone', 'ph', 'pm1', 'pm10', 'pm25', 'power_factor', 'power', 'precipitation', 'precipitation_intensity', 'pressure', 'reactive_power', 'signal_strength', 'sound_pressure', 'speed', 'sulphur_dioxide', 'temperature', 'volatile_organic_compounds', 'volatile_organic_compounds_parts', 'voltage', 'volume', 'volume_storage', 'volume_flow_rate', 'water', 'weight', 'wind_speed' for dictionary value @ data['device_class']' when processing MQTT discovery message topic: 'homeassistant/sensor/xpeng_smart_charging_enabled/config', message: '{'state_topic': 'xpeng/smart_charging_policy/isEnabled', 'device_class': 'switch', 'unique_id': 'xpeng_smart_charging_enabled', 'device': {'identifiers': ['xpeng'], 'model': 'Xpeng Model', 'manufacturer': 'Xpeng', 'name': 'Xpeng'}, 'name': 'Xpeng Smart Charging Enabled'}'

**Comment by Fannangir** [comment]
Am I the only one having these errors? :)

**Comment by OldSmurf56** [comment]
Yeah - so it would appear - at least based on the feedback I have received, users are successfully deploying my solution

**Comment by Fannangir** [comment]
The solution is working great!
Thank you very much.

I guess the reason is that the type of the sensor is not supported by HA (the type undefined, or defined as one of not recognized by HA).
